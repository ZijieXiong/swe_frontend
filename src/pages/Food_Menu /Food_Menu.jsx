import React from 'react';
import {useHistory} from 'react-router-dom';

export default function Home(){
  const history = useHistory();

  function navigateToPage(path) {
    history.push(path);
  }

  return (
    <div className="content">


      <ul>
      <li><a class="active" href="https://www.google.com/">Home</a></li>
      <li><a href="#news">Food menu</a></li>
      <li><a href="#contact">Drink Menu</a></li>
      <li><a href="www.google.com">Delivery!</a></li>
     </ul>
     



      <h1> Pancake Flippers </h1>
      <h1>ðŸ¤— Welcome!</h1>
      <button
        onClick={() => navigateToPage('/foodmenu')}
        className="page-button"
      >
        Food Menu
      </button>
      <button
        onClick={() => navigateToPage('/users')}
        className="page-button"
      >
        Drink Menu
      </button> 
      <button 
        onClick = {() => navigateToPage('/rooms')}
        className= "page-button"
      >
        Delivery 
      </button>
    </div>
  );
};

