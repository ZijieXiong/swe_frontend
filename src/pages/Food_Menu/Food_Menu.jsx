import React from 'react'
//import {useEffect} from 'react'
//Utilizing foodmenu_css 
import axios from 'axios' //to use axios run "npm install axios"  in terminal
import { useState, useEffect } from 'react';



export default function Food_Menu({foodName}) {

    const [foodType, setFoodType] = useState([])
    useEffect(() => {
        // adding event listeners on mount here
        axios.get('https://swe-temp.herokuapp.com/food_menu/type')
        .then((res) => {
            if (res.data) {
                const formattedData = Object.values(res.data)
                  setFoodType(formattedData);
             }
        })
        
        }, [] );

    const[app_fooditems, setAppFoodItems] = useState(undefined)
    const[ent_fooditems, setEntFoodItems] = useState(undefined)
    const[des_fooditems, setDesFoodItems] = useState(undefined)

    useEffect(() => {
        axios.get('https://swe-temp.herokuapp.com/food_menu/list')
        .then((res) => {

            //if(res.data) {
           // const beverages = Object.value(res.data.Alcoholic.Beer)
           //const beverages = Object.entries(res.data.Alcoholic.Beer)
           const app_food = []
	   const ent_food = []
	   const des_food = []
           //console.log((res.data.Beer))
           // do foreach
           Object.keys(res.data.Appetizer).forEach(food=>{
           app_food.push({foodName: res.data.Appetizer[food].foodName, price: res.data.Appetizer[food].price, description: res.data.Appetizer[food].description})
           console.log(app_food) })

	   Object.keys(res.data.Entree).forEach(food=>{
	   ent_food.push({foodName: res.data.Entree[food].foodName, price: res.data.Entree[food].price, description: res.data.Entree[food].description})
	   console.log(ent_food) })

	   Object.keys(res.data.Dessert).forEach(food=>{
           des_food.push({foodName: res.data.Dessert[food].foodName, price: res.data.Dessert[food].price, description: res.data.Dessert[food].description})
	   console.log(des_food) })

	   setAppFoodItems(app_food)
	   setEntFoodItems(ent_food)
	   setDesFoodItems(des_food)
        })
        

    }, [])

    return (

        <>

	<div>
        {foodType.map((type) => {
	   
	   let foodItems;
	   
	   switch (type.typeName) {
	      case "Appetizer":
		 foodItems = app_fooditems
	         break;

	      case "Entree":
		 foodItems = ent_fooditems
	         break;

	      case "Dessert": 
		 foodItems = des_fooditems
		 break;
	   }

	return (
        
	    <div>
           <p key={type.typeName}> {type.typeName} </p>
	        <div>
	        <section class = "grid_container">
	{foodItems && foodItems.map((item) => (
		<div key={item.foodName}>
		   <div className = "fooditems">

		   <h1 class = "foodName"> {item.drinkName}</h1>
		   <p class = "price"> Description: {item.description}</p>
		   <p class = "price"> Price: ${item.price} </p>

		</div>
		</div>

        	))}
		</section>
    </div>
	    </div>
    )
    })}
    </div>
    </>
    )
}
