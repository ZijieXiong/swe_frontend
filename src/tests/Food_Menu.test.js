import React from 'react'
import renderer from 'react-test-renderer'
import TestRenderer from 'react-test-renderer';

//npm install --save-dev react-test-renderer

import Food_Menu from '../pages/Food_Menu/Food_Menu.jsx'
it('renders correctly', () => {
  const tree = renderer
    .create(<Food_Menu/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
