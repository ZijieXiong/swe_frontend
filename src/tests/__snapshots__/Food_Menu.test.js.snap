// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="food_menu"
>
  <div
    class="extra_space"
  />
</div>
`;
